tests/test_views.py

```diff
@@ -119,12 +119,44 @@ def test_can_be_anonymous_for_public_project(self, mock_property):
         auth = Auth(user=self.user, private_key=anonymous_link.key)
         assert_true(has_anonymous_link(self.project, auth))
 
-    def test_has_private_link_key(self):
+    @mock.patch('framework.sessions.get_node_id')
+    def test_has_private_link_key(self, mock_get_node_id):
+        mock_get_node_id.return_value = self.project
         res = self.app.get(self.project_url, {'view_only': self.link.key})
         assert_equal(res.status_code, 200)
 
+    def test_has_bad_private_link_key_redirects_properly(self):
+        res = self.app.get(self.project_url, {'view_only': 'not_valid_link'})
+        assert_is_redirect(res)
+        res = res.follow(expect_errors=True)
```

Comment:
Here and other places in this file, the `expect_errors` flag should only need to be present if errors are expected -- that is, if the response status code is in the 4XX-5XX range.